cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -1172,8 +1179,18 @@ def run_query(self, off_line=False, disp_conf=None):
                 if job_monitor is None:
                     job_monitor = job.monitor
             else:
-
+                query_name = self.instrument.get_product_query_name(product_type)
+                if not self.public:
+                    roles = []
+                    if self.token is not None and self.decoded_token is not None:
+                        roles = self.get_token_roles()
+                    # assess the permissions for the query execution
+                    try:
+                        self.instrument.check_instrument_query_role(query_name, roles)
```

Comment:
it will need par_dict I think, to make filtering by them

Reply:
You mean to have more information for evaluate the permission of the user over a certain query?

Reply:
yes, we need to also filter by parameters of the specific request.